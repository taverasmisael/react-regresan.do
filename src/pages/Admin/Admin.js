import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import Typography from 'material-ui/Typography'

import Routes from './routes'
export default class AdminPage extends PureComponent {

  render() {
    return (
      <React.Fragment>
        <Typography type="subheading">Admin</Typography>
        <main>
          <Link to="/admin/">Overview?</Link>
          <Link to="/admin/branches/">Branchess?</Link>
          <Routes base={this.props.match.path} />
        </main>
      </React.Fragment>
    )
  }
}
