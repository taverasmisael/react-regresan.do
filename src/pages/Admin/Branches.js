import React, { PureComponent } from 'react'

export default class Branches extends PureComponent {
  render() {
    return (
      <React.Fragment>
        Branches
      </React.Fragment>
    )
  }
}
