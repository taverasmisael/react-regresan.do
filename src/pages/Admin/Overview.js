import React, { PureComponent } from 'react'

export default class OverviewPage extends PureComponent {
  render() {
    return (
      <React.Fragment>
        Overview
      </React.Fragment>
    )
  }
}
