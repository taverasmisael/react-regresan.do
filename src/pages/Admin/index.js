import OverviewPage from './Overview'
import BranchesPage from './Branches'
import AdminPage from './Admin'

export { OverviewPage, BranchesPage, AdminPage }
