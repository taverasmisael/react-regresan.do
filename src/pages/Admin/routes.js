import React from 'react'
import PropTypes from 'prop-types'
import { Switch, Route } from 'react-router-dom'

import OverviewPage from './Overview'
import BranchesPage from './Branches'

const Routes = ({ base = '/admin' }) => (
  <Switch>
    <Route exact path={base} component={OverviewPage} />
    <Route exact path={`${base}/branches`} component={BranchesPage} />
  </Switch>
)

Routes.propTypes = {
  base: PropTypes.string.isRequired
}

export default Routes
