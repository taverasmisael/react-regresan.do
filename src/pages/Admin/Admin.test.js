import React from 'react'
import { shallow } from 'enzyme'
import Admin from './Admin'

describe('routes', () => {
  describe('Pages/Admin', () => {
    it('renders without crashing', () => {
      shallow(<Admin match={{ path: '/admin' }} />)
    })
    it('matches snapshot', () => {
      expect(shallow(<Admin match={{ path: '/admin' }} />)).toMatchSnapshot()
    })
  })
})
