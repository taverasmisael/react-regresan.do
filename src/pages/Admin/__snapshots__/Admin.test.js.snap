// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes Pages/Admin matches snapshot 1`] = `
<React.Fragment>
  <WithStyles(Typography)
    type="subheading"
  >
    Admin
  </WithStyles(Typography)>
  <main>
    <Link
      replace={false}
      to="/admin/"
    >
      Overview?
    </Link>
    <Link
      replace={false}
      to="/admin/branches/"
    >
      Branchess?
    </Link>
    <Routes
      base="/admin"
    />
  </main>
</React.Fragment>
`;
